lightning/src/ln/functional_test_utils.rs

```diff
@@ -2768,8 +2768,8 @@ pub fn do_pass_along_path<'a, 'b, 'c>(args: PassAlongPathArgs) -> Option<Event>
 							},
 						}
 						assert_eq!(*amount_msat, recv_value);
-						assert!(node.node.list_channels().iter().any(|details| details.channel_id == via_channel_id.unwrap()));
-						assert!(node.node.list_channels().iter().any(|details| details.user_channel_id == via_user_channel_id.unwrap()));
+						assert!(node.node.list_channels().iter().any(|details| details.channel_id == via_channel_ids.last().unwrap().unwrap()));
+						assert!(node.node.list_channels().iter().any(|details| details.user_channel_id == via_user_channel_ids.last().unwrap().unwrap()));
```

Comment:
For completeness, could we iterate through the event channel ids list and make sure all of them are present in `list_channels`?

Reply:
Updated in [pr3655.02](https://github.com/shaavan/rust-lightning/commits/pr3655.02)
Thanks for the pointer!